### PR TITLE
docs: improve README

### PR DESCRIPTION
**Pull request by dcoa**
This PR has the goal of improving the plugin description throughout the README.

Also, I would like to change the repository description for the current introduction:

> Tutor plugin that enables execution of untrusted code in secure sandboxes using an external service based on the codejail library.

[JIRA](https://edunext.atlassian.net/browse/DS-971)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,18 +11,23 @@ Tutor plugin that enables execution of untrusted code in secure sandboxes using
 Installation
 ------------
 
+To install the latest version run:
+
 .. code-block:: bash
 
     pip install git+https://github.com/edunext/tutor-contrib-codejail
 
+You can install a specific version by adding the tag at the end, e.g.: ``#v17.0.0``.
```
Comment (mariajgrimaldi) [inline README.rst]: What do you think about this?
```suggestion
You can install a specific version by adding the tag, branch, or commit:

.. code-block:: bash

    pip install git+https://github.com/edunext/tutor-contrib-codejail@v17.0.0
```
https://pip.pypa.io/en/latest/cli/pip_install/#examples

File: README.rst [hunk 2]
```diff
@@ -37,10 +43,10 @@ Finally, the platform can be run as usual:
 Configuration
 -------------
 
-For some of these configurations to work correctly, the codejail image must be built again. Command to build codejail: ``tutor images build codejail``.
+To customize the configuration, update the following settings in Tutor:
 
 - ``CODEJAIL_APPARMOR_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejail_apparmor_loader:latest``)
-- ``CODEJAIL_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejailservice:14.0.0``)
+- ``CODEJAIL_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejailservice:<_version_>``)
```
Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
- ``CODEJAIL_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejailservice:{__version__} ``) 
```
So it's the same as in the plugin.py file defaults.

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: README.rst [hunk 3]
```diff
@@ -10,18 +11,27 @@ Tutor plugin that enables execution of untrusted code in secure sandboxes using
...
 Usage
 -----
+Enable the plugin with:
 
 .. code-block:: bash
 
     tutor plugins enable codejail
 
-Then, you will have to install the "docker-edx-sandbox" apparmor profile on your host:
+Install the "docker-edx-sandbox" AppArmor profile on your host:
```
Comment (MoisesGSalas) [inline README.rst]:
```suggestion
Run the initialization jobs to install the required AppArmor profile on your host:
```

The name of the profile changes depending on the python version used by the sandbox (e.g `docker-edx-sandbox-3.8.6`).

Can we also add a note that the profile must be loaded again if the host is rebooted?

File: README.rst (hunk 2 above)
Comment (dcoa) [reply README.rst]: I use YML syntax because using single brackets made the quality test fail.

File: README.rst [hunk 4]
```diff
@@ -34,13 +44,15 @@ Finally, the platform can be run as usual:
...
+
 Configuration
 -------------
 
-For some of these configurations to work correctly, the codejail image must be built again. Command to build codejail: ``tutor images build codejail``.
+To customize the configuration, update the following settings in Tutor:
 
 - ``CODEJAIL_APPARMOR_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejail_apparmor_loader:latest``)
-- ``CODEJAIL_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejailservice:14.0.0``)
+- ``CODEJAIL_DOCKER_IMAGE``: (default: ``docker.io/ednxops/codejailservice:{{__version__}}``)
```
Comment (mariajgrimaldi) [inline README.rst:55]: can you improve the formatting of the CODEJAIL_EXTRA_PIP_REQUIREMENTS code block?

Comment (dcoa) [reply README.rst:55]: I made a change, could you check please?

File: README.rst [hunk 5]
```diff
@@ -34,23 +44,27 @@ Finally, the platform can be run as usual:
...
     
-    .. code-block:: yaml
+.. code-block:: yaml
 
-        CODEJAIL_EXTRA_PIP_REQUIREMENTS:
-        - pybryt
+    CODEJAIL_EXTRA_PIP_REQUIREMENTS:
+    - pybryt
+        
+**NOTE:** For some configurations to work, the codejail image must be rebuilt. Command to rebuild codejail: ``tutor images build codejail``.
```
Comment (mariajgrimaldi) [inline README.rst]: Do we know which configurations merit a new build? Can we add them here?

Comment (MoisesGSalas) [reply README.rst]: I think most people don't build this image, so it should rather be framed as what would require a custom build:

1. Additional requirements in the sandbox via `CODEJAIL_EXTRA_PIP_REQUIREMENTS`.
2. A different version of python for the sandbox environment via `CODEJAIL_SANDBOX_PYTHON_VERSION`.
3. Custom version of edx-platform that changes the contents of [`requirements/edx-sandbox`](https://github.com/openedx/edx-platform/tree/open-release/redwood.master/requirements/edx-sandbox).

One plugin that includes a section like this is [`tutor-ecommerce`](https://github.com/overhangio/tutor-ecommerce?tab=readme-ov-file#image-customisation).

File: README.rst (hunk 3 above)
Comment (dcoa) [reply README.rst]: Implemented :)

File: README.rst (hunk 5 above)
Comment (dcoa) [reply README.rst]: Implemented :)

File: README.rst [hunk 6]
```diff
@@ -1,7 +1,8 @@
 Codejail plugin for `Tutor`_
 ============================
 
-Tutor plugin that enables execution of untrusted code in secure sandboxes using an external `service`_ based on the `codejail`_ library.
+Tutor plugin that configures and runs Codejail using a REST API service. Codejail allows for the
```
Comment (MoisesGSalas) [inline README.rst]: Overall I think is good to go, but if you have time can you include the link to the edunext/codejailservice repository?

**Review by MoisesGSalas — APPROVED** [review]

**Review by mariajgrimaldi — APPROVED** [review]

File: README.rst (hunk 6 above)
Comment (dcoa) [reply README.rst]: Sure

**Pull request merged**